### ISSUE TITLE
problem during train ESCPN with tensorflow

### ISSUE THREAD

**Issue by huang-junhong** [huang-junhong/ESPCN#1]
I try to wirte ESCPN with Tensorflow, but something strange
    1.when strt it's take long time befor train
       I found befor train the optimizer cost long
    2.wait for train after the first train,it can print the result then the program die.
there are my ESCPN_process and recover_to_sr_img code:
`def ESPCN_process(input_img,reuse=False):
    with tf.variable_scope('ESPCN',reuse=reuse):

        #reshape tensor
        lr_img=tf.reshape(input_img,[-1,7,7,1])
        #weigths initial
        conv1_weights=tf.get_variable(name='conv1_weights',shape=[3,3,1,4],initializer=tf.contrib.layers.xavier_initializer_conv2d(),trainable=True)
        conv2_weights=tf.get_variable(name='conv2_weights',shape=[3,3,4,8],initializer=tf.contrib.layers.xavier_initializer_conv2d(),trainable=True)
        conv3_weights=tf.get_variable(name='conv3_weights',shape=[3,3,8,16],initializer=tf.contrib.layers.xavier_initializer_conv2d(),trainable=True)
        #bias initial
        conv1_bias=tf.get_variable(name='conv1_bias',shape=[4],initializer=tf.contrib.layers.xavier_initializer_conv2d(),trainable=True)
        conv2_bias=tf.get_variable(name='conv2_bias',shape=[8],initializer=tf.contrib.layers.xavier_initializer_conv2d(),trainable=True)
        conv3_bias=tf.get_variable(name='conv3_bias',shape=[16],initializer=tf.contrib.layers.xavier_initializer_conv2d(),trainable=True)

        #1st conv
        conv1=tf.nn.relu(tf.nn.conv2d(lr_img,conv1_weights,strides=[1,1,1,1],padding='SAME')+conv1_bias)
        conv1=tf.layers.batch_normalization(conv1,training=True)

        #2nd conv
        conv2=tf.nn.relu(tf.nn.conv2d(conv1,conv2_weights,strides=[1,1,1,1],padding='SAME')+conv2_bias)
        conv2=tf.layers.batch_normalization(conv2,training=True)

        #3rd conv
        conv3=tf.nn.relu(tf.nn.conv2d(conv2,conv3_weights,strides=[1,1,1,1],padding='SAME')+conv3_bias)
        
        if conv3.shape[0].value>3:
            conv3=recover_to_sr(conv3,batch_size)
        else:
            conv3=recover_to_sr(conv3,sample_size)

        return conv3

def recover_to_sr(conv_res,batch_size):
    srss=conv_res
    for batch_i in range(batch_size):
        print('recover '+str(batch_i)+'th')
        sr=conv_res
        for i in range(7):
            for j in range(7):
                if j==0:
                    #新的一行开始了
                    ta0=conv_res[batch_i][i][j]
                    ta0=tf.reshape(ta0,[4,4])
                else:
                    ta1=conv_res[batch_i][i][j]
                    ta1=tf.reshape(ta1,[4,4])
                    ta0=tf.concat([ta0,ta1],1)
                    if i==0 and j==6:
                        sr=ta0
                    elif j==6:
                        sr=tf.concat([sr,ta0],0)
        sr=tf.reshape(sr,[1,784])
        if batch_i==0:
            srss=sr
        else:
            srss=tf.concat([srss,sr],0)
    return srss

batch_holder=tf.placeholder(tf.float32,[batch_size,784])
lri=tf.placeholder(tf.float32,[batch_size,49])
ori=tf.placeholder(tf.float32,[batch_size,784])

slri=tf.placeholder(tf.float32,[sample_size,49])
sori=tf.placeholder(tf.float32,[sample_size,784])

OR_img=get_or_img(batch_holder)

sri=ESPCN_process(lri,reuse=False)
ssri=ESPCN_process(slri,reuse=True)

trainble=tf.trainable_variables()

Loss_mse=tf.reduce_mean(tf.square(ori-sri))

mse_opt=tf.train.AdamOptimizer(learning_rate).minimize(Loss_mse,var_list=trainble)

losses=[]
`
